### PR TITLE
Creates wire-grpc-server

### PR DESCRIPTION
**Pull request by oldergod**
Also introduces GrpcChannel to be used by generated server code.

### CODE REVIEW

File: wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import java.io.Closeable
+
+interface GrpcSendChannel<T> : Closeable {
```
Comment (swankjesse) [inline wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: Should these go in grpc-common to give us the option for blocking clients?

Comment (oldergod) [reply wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: Why not? We'd have to support it in the runtime as well. Our `GrpcClient` assumes coroutines to create the service.

You'd add something like in the plugin config?
```kotlin
    var blockingServices: Boolean
    var singleMethodServices: Boolean
/!\ var kotlinCoroutine: Boolean /!\
```

Comment (swankjesse) [reply wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: I think it's just blockingServices=true to use these.

Comment (Egorand) [inline wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: nit: `<in T>`?

File: wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  fun send(message: T)
+}
+
+interface GrpcReceiveChannel<T> : Closeable {
```
Comment (Egorand) [inline wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: nit: `<out T>`?

File: wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  fun receiveOrNull(): T?
+}
+
+fun <T> GrpcReceiveChannel<T>.consumeEachAndClose(block: (T) -> Unit) {
```
Comment (Egorand) [inline wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: nit: `inline`

File: wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt (hunk 1 above)
Comment (oldergod) [reply wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: `blockingServices` makes it so we don't use `suspend` for a single request/response API. Others API are non-blocking by default.
Why would `blockingServices=true` affect the non-blocking APIs signature?

Comment (swankjesse) [reply wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: I think there's two independent features:

 * blocking/suspending
 * single interface/many interfaces

We could build blocking clients and suspending servers. Just don't need those yet.

Comment (oldergod) [reply wire-grpc-server/src/main/java/com/squareup/wire/GrpcChannel.kt]: Ok, then maybe we don't need the `wire-grpc-server` for now. I'll move the GrpcChannels into our `wire-grpc` module.

**Pull request closed without merging**